lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -156,6 +183,463 @@ def validateFASTA(self, ctx, params):
+        with open(file_path, 'w') as fhandle:
+            self.log('downloading reads file: ' + str(file_path))
+            r = requests.get(node_url + '?download', stream=True,
+                             headers=headers)
+            self.check_shock_response(r)
+            for chunk in r.iter_content(1024):
+                if not chunk:
+                    break
+                fhandle.write(chunk)
+        return file_path, gzipped
+
+    def make_ref(self, object_info):
+        return str(object_info[6]) + '/' + str(object_info[0]) + \
+            '/' + str(object_info[4])
+
+    def check_reads(self, reads):
+        info = reads['info']
+        obj_ref = self.make_ref(info)
+        obj_name = info[1]
+
+        # Might need to do version checking here.
+        module_name, type_name = info[2].split('-')[0].split('.')
+        if (module_name not in self.MODULE_NAMES or
+                type_name not in self.TYPE_NAMES):
+            types = []
+            for mod in self.MODULE_NAMES:
+                for type_ in self.TYPE_NAMES:
+                    types.append(mod + '.' + type_)
+            raise ValueError(('Invalid type for object {} ({}). Supported ' +
+                              'types: {}').format(obj_ref, obj_name,
+                                                  ' '.join(types)))
+        return (type_name == self.SINGLE_END_TYPE,
+                module_name == self.KBASE_FILE)
+
+    def copy_field(self, source, field, target):
+        target[field] = source.get(field)
+
+    # this assumes that the FASTQ file is properly formatted, which it should
+    # be if it's in KBase. Credit:
+    # https://www.biostars.org/p/19446/#117160
+    def deinterleave(self, filepath, fwdpath, revpath):
+        self.log('Deinterleaving file {} to files {} and {}'.format(
+            filepath, fwdpath, revpath))
+        with open(filepath, 'r') as s:
+            with open(fwdpath, 'w') as f, open(revpath, 'w') as r:
+                for i, line in enumerate(s):
+                    if i % 8 < 4:
+                        f.write(line)
+                    else:
+                        r.write(line)
+
+    # this assumes that the FASTQ files are properly formatted and matched,
+    # which they should be if they're in KBase. Credit:
+    # https://sourceforge.net/p/denovoassembler/ray-testsuite/ci/master/tree/scripts/interleave-fastq.py
+    def interleave(self, fwdpath, revpath, targetpath):
+        self.log('Interleaving files {} and {} to {}'.format(
+            fwdpath, revpath, targetpath))
+        with open(targetpath, 'w') as t:
+            with open(fwdpath, 'r') as f, open(revpath, 'r') as r:
+                while True:
+                    line = f.readline()
+                    # since FASTQ cannot contain blank lines
+                    if not line or not line.strip():
+                        break
+                    t.write(line.strip() + '\n')
```

Comment:
Currently the uploader allows blank lines in fq. Either this needs to handle & ignore blank lines or the uploader needs to fix them and reupload the files.

Same for the deinterleave fn above.